### PR TITLE
“Fix Middlewoes”, or “OMG, It Was You the Entire Time”

### PR DESCRIPTION
**Pull request by keirog**
So… it turns out we’ve been mounting all routes to both the apiRoute *and* authRoute routers. 😱 

This PR corrects the problem and attempts to make the app.js file more reflective of what happens at runtime.

I highly recommend viewing this PR one commit at a time 😬 

cat makes a discovery

### CODE REVIEW

File: server/middleware/auth/public.js [hunk 1]
```diff
@@ -2,13 +2,18 @@
...
 const logger = require('@financial-times/n-logger').default;
 const whitelist = [
-	/^\/myft\/product-tour(?!-get-started)/
+	/^\/myft\/product-tour(?!-get-started)/,
+	/^\/(__)?myft\/list\/([^/]+)$/
 ];
 
 module.exports = (req, res, next) => {
-	logger.info('In public middleware', { event: 'BUG_ADVENTURE', url: req.url, originalUrl: req.originalUrl });
+	logger.info('In public middleware', { event: 'BUG_ADVENTURE', method: req.method, url: req.url, originalUrl: req.originalUrl });
```
Comment (phamann) [inline server/middleware/auth/public.js]: Do we still need this debug log?

**Comment by phamann** [comment]
Woohoo! But the suspense is killing me, did it fix your core journey bug?

**Comment by phamann** [comment]
👍 💯

File: server/middleware/auth/public.js (hunk 1 above)
Comment (keirog) [inline server/middleware/auth/public.js]: Just leaving this in until everything is confirmed fixed in prod

**Comment by keirog** [comment]
I‘m *really* hoping so… I can‘t be sure until merging

**Comment by keirog** [comment]
(For those following along, it didn‘t fix it but did make it _much_ easier to find out was going on)

**Pull request merged**